apps/src/storage/dataBrowser/dataVisualizer/CrossTabChart.jsx

```diff
@@ -1,146 +1,188 @@
+    props.records,
+    props.numericColumns,
+    props.selectedColumn1,
+    props.selectedColumn2
+  );
+  const numericValues = [];
 
-class CrossTabChart extends React.Component {
-  static propTypes = {
-    records: PropTypes.array.isRequired,
-    numericColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
-    chartTitle: PropTypes.string,
-    selectedColumn1: PropTypes.string,
-    selectedColumn2: PropTypes.string
-  };
+  chartData.forEach(record => {
+    Object.entries(record).forEach(entry => {
+      let key = entry[0];
+      let value = entry[1];
+      if (typeof value === 'number' && key !== props.selectedColumn1) {
+        numericValues.push(value);
+      }
+    });
+  });
 
-  /**
-   * @param {Array.<Object>} records
-   * @param {string} rowName
-   * @param {string} columnName
-   */
-  createPivotTable = (records, rowName, columnName) => {
-    let countMap = {};
+  // Our goal is uniform column widths, for all columns except the first one.
+  // There are a few steps that lead to successful layout here.
+  // 1. The table width is 100% to maximize available space.
+  // 2. The first column uses `white-space: nowrap` so it won't shrink too
+  //    small for its contents.
+  // 3. We set the rest of the columns to have the same percentage-width,
+  //    adding up to 99% of the table width (which is more than available space)
+  //    so after setting the first column width, the remaining columns
+  //    shrink to fit evenly.
+  const columnWidth = 99 / (columns.length - 1) + '%';
 
-    // Find all values in columnName - these will be the columns of the pivot table
-    const pivotedColumns = new Set(records.map(record => record[columnName]));
+  const min = Math.min(...numericValues);
+  const max = Math.max(...numericValues);
 
-    // Count occurrences of each row/column pair
-    records.forEach(record => {
-      let key = record[rowName];
-      let value = record[columnName];
-      if (!countMap[key]) {
-        countMap[key] = {[rowName]: key};
-        pivotedColumns.forEach(column => (countMap[key][column] = 0));
-      }
-      countMap[key][value]++;
-    });
+  return (
+    <div id={CROSS_TAB_CHART_AREA} style={wrapperStyle}>
+      <h1 style={chartTitleStyle}>{props.chartTitle}</h1>
+      <table style={tableStyle}>
+        <tbody>
+          <tr>
+            <td>&nbsp;</td>
+            <td
+              style={{...topCellStyle, ...axisTitleStyle}}
+              colSpan={columns.length - 1}
+            >
+              {props.selectedColumn2}
+            </td>
+          </tr>
+          <tr>
+            {columns.map((column, i) => (
+              <td
+                key={column}
+                style={
+                  i === 0 ? {...leftCellStyle, ...axisTitleStyle} : topCellStyle
+                }
+              >
+                {column}
+              </td>
+            ))}
+          </tr>
+          {chartData.map((record, i) => (
+            <tr key={i}>
```

Comment:
Probably doesn't matter much in this case, but I think using indices as key is an antipattern for react. Consider using `record.id` instead which is guaranteed unique.

Reply:
Oh, I totally forgot about this! [Good description of the reason index-as-key is an antipattern here](https://medium.com/@vraa/why-using-an-index-as-key-in-react-is-probably-a-bad-idea-7543de68b17c). Thanks for catching this!